src/lib/items/Item.js

```diff
@@ -1,571 +1,615 @@
-    // TODO: conditionals are really ugly.  could use Fragment if supporting React 16+ but for now, it'll
-    // be ugly
-    return (
-      <div
-        {...this.props.item.itemProps}
-        key={this.itemId}
-        ref={el => (this.item = el)}
-        className={classNames}
-        title={this.itemDivTitle}
-        onMouseDown={this.onMouseDown}
-        onMouseUp={this.onMouseUp}
-        onTouchStart={this.onTouchStart}
-        onTouchEnd={this.onTouchEnd}
-        onDoubleClick={this.handleDoubleClick}
-        onContextMenu={this.handleContextMenu}
-        style={style}
-      >
-        {this.props.useResizeHandle && showInnerContents ? (
-          <div ref={el => (this.dragLeft = el)} className="rct-drag-left" />
-        ) : (
-          ''
-        )}
-
-        {showInnerContents ? (
-          <div
-            className="rct-item-content"
-            style={{
-              maxWidth: `${dimensions.width}px`
-            }}
-          >
-            {this.renderContent()}
-          </div>
-        ) : (
-          ''
-        )}
-
-        {this.props.useResizeHandle && showInnerContents ? (
-          <div ref={el => (this.dragRight = el)} className="rct-drag-right" />
-        ) : (
-          ''
-        )}
-      </div>
-    )
-  }
-}
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import interact from 'interactjs'
+import moment from 'moment'
+
+import { _get, deepObjectCompare } from '../utility/generic'
+import { composeEvents } from '../utility/events'
+
+export default class Item extends Component {
+  // removed prop type check for SPEED!
+  // they are coming from a trusted component anyway
+  // (this complicates performance debugging otherwise)
+  static propTypes = {
+    canvasTimeStart: PropTypes.number.isRequired,
+    canvasTimeEnd: PropTypes.number.isRequired,
+    canvasWidth: PropTypes.number.isRequired,
+    order: PropTypes.number,
+    minimumWidthForItemContentVisibility: PropTypes.number.isRequired,
+
+    dragSnap: PropTypes.number,
+    minResizeWidth: PropTypes.number,
+    selected: PropTypes.bool,
+
+    canChangeGroup: PropTypes.bool.isRequired,
+    canMove: PropTypes.bool.isRequired,
+    canResizeLeft: PropTypes.bool.isRequired,
+    canResizeRight: PropTypes.bool.isRequired,
+
+    keys: PropTypes.object.isRequired,
+    item: PropTypes.object.isRequired,
+
+    onSelect: PropTypes.func,
+    onDrag: PropTypes.func,
+    onDrop: PropTypes.func,
+    onResizing: PropTypes.func,
+    onResized: PropTypes.func,
+    onContextMenu: PropTypes.func,
+    itemRenderer: PropTypes.func,
+
+    itemProps: PropTypes.object,
+    canSelect: PropTypes.bool,
+    topOffset: PropTypes.number,
+    dimensions: PropTypes.object,
+    groupTops: PropTypes.array,
+    useResizeHandle: PropTypes.bool,
+    moveResizeValidator: PropTypes.func,
+    onItemDoubleClick: PropTypes.func
+  }
+
+  static defaultProps = {
+    selected: false,
+    itemRenderer: ({

```

Comment:
Can you pull this default `itemRenderer` out into a separate function outside of the component? Just to be clean and explicit.